src/rngs/adapter/reseeding.rs

```diff
@@ -220,6 +260,48 @@ impl<R, Rsdr> CryptoRng for ReseedingCore<R, Rsdr>
 where R: BlockRngCore + SeedableRng + CryptoRng,
       Rsdr: RngCore + CryptoRng {}
 
+
+// Fork protection
+//
+// We implement fork protection on Unix using `pthread_atfork`.
+// When the process is forked, we increment `RESEEDING_RNG_FORK_COUNTER`.
+// Every `ReseedingRng` stores the last known value of the static in
+// `fork_counter`. If the cached `fork_counter` is less than
+// `RESEEDING_RNG_FORK_COUNTER`, it is time to reseed this RNG.
+//
+// If reseeding fails, we don't deal with this by setting a delay, but just
+// don't update `fork_counter`, so a reseed is attempted a soon as possible.
+#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
+static RESEEDING_RNG_FORK_COUNTER: AtomicUsize = ATOMIC_USIZE_INIT;
```

Comment:
You could use `#[cfg(..)] mod fork { .. }` so that the `cfg` stuff doesn't need to be repeated so much. Not important.

Reply:
Much cleaner :smile: